File: sdk/typescript/smoke-test.mjs
import { Once } from "./dist/index.js";
import { randomUUID } from "node:crypto";

const once = new Once();

const operationId =
  "sdk-smoke-" + randomUUID().replaceAll("-", "");

console.log("Operation:", operationId);

const first = await once.execute({
  operationId,
  provider: "blind_test",
  action: {
    type: "typescript_sdk_smoke_test"
  }
});

console.log("\nFIRST CALL");
console.log(first);

const second = await once.execute({
  operationId,
  provider: "blind_test",
  action: {
    type: "typescript_sdk_smoke_test"
  }
});

console.log("\nEXACT RETRY");
console.log(second);

const truth = await once.truth(operationId);

console.log("\nTRUTH");
console.log(truth);

if (truth.side_effects !== 1) {
  throw new Error(
    `SAFETY FAILURE: expected 1 side effect, got ${truth.side_effects}`
  );
}

console.log("\nSDK SAFETY TEST PASSED");
console.log("Exactly one provider side effect.");

File: sdk/python/smoke_test.py
import sys
import uuid

sys.path.insert(0, r".\sdk\python\src")

from once_agent import Once

once = Once()

operation_id = "python-sdk-" + uuid.uuid4().hex

print("Operation:", operation_id)

first = once.execute(
    operation_id=operation_id,
    provider="blind_test",
    action={
        "type": "python_sdk_smoke_test"
    },
)

print("First:", first.get("result"))

retry = once.execute(
    operation_id=operation_id,
    provider="blind_test",
    action={
        "type": "python_sdk_smoke_test"
    },
)

print("Retry:", retry.get("result"))

truth = once.truth(operation_id)

print("State:", truth.get("ledger_state"))
print("Side effects:", truth.get("side_effects"))

if truth.get("side_effects") != 1:
    raise RuntimeError("SAFETY FAILURE")

print("PYTHON SDK SAFETY TEST PASSED")
